### PR TITLE
[MM-36421] Replace server tabs with dropdown menu

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
In preparation for new designs around the Desktop App wrt. Focalboard and Playbook, we've changed the server selection to use a custom dropdown, allowing the tabs to do something else.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36421

#### Screenshots
image

#### Release Note
```release-note
Changed server selection to use a dropdown instead of tabs.
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
@matthewbirtch Let me know if you would like a build generated for this specific change or if you'd rather wait until it's merged and pick up a daily build.

**Comment by matthewbirtch** [comment]
> @matthewbirtch Let me know if you would like a build generated for this specific change or if you'd rather wait until it's merged and pick up a daily build.

A build would be great @devinbinnie!

**Review by jwilander — APPROVED** [review]
Looks good 👍 Two non-blocking comments

File: src/main/preload/dropdown.js
Comment (devinbinnie) [reply src/main/preload/dropdown.js:18]: We usually keep these here in case we need them, just to make sure the preload script has loaded. The dev tools isn't accessible by the end user for this browser view so the user will never see it. I suppose we can remove it if need be.

File: src/main/views/teamDropdownView.ts
Comment (devinbinnie) [reply src/main/views/teamDropdownView.ts:22]: We still use "Team" in the desktop app to denote a server, even though it's wrong :P
I just did this for consistency. @Willyfrog do we do a mass rename here?
I feel like it might be worth doing this as a cleanup step en masse later on.

Comment (Willyfrog) [reply src/main/views/teamDropdownView.ts:22]: I'm always in the verge of doing it, so maybe it is the right time to do it :)

**Review by Willyfrog — APPROVED** [review]

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Thanks @devinbinnie. Nice work. All minor things from me:
1. Can we use Open Sans for everything with these new controls? I see it's being used for the server dropdown menu title, but not the menu items themselves. It also looks like the dropdown button itself is not using Open Sans.
2. If the count on the dropdown is >99, can we make it show 99+? 
image
3. Same as above for each of the menu items - can they show 99+ if  the count is > 99?
4. Now that I'm seeing it in the product, I'm wondering if the height of each menu item in this servers menu is too tall - it's causing a lot of extra white space. I think we could get away with 36px height rather than 40px in height. I've updated Figma to reflect this.
5. For the active state of the dropdown button, can we update the border around the mention badge to be the same as the background color?
image

6. Can you confirm that all the mention badges are using the bold weight of Open Sans? (700)
7. When I click one of the items in the dropdown and then open the dropdown again, the previously clicked item displays the active state background color. It seems like the state is not reset. I actually don't mind this, except for the case of 'Add a server'. It feels a bit odd.
8. For the dropdown menu item hover state, the trash icon isn't quite aligned properly when there is no mention badge
image
9. The divider lines in the menu look like they are 2px rather than 1px

File: src/main/views/teamDropdownView.ts
Comment (jwilander) [reply src/main/views/teamDropdownView.ts:22]: 0/5 on doing it now or later but something we should do we eventually

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
A few other things I noticed @devinbinnie 

1. The number in the mention badge on the dropdown button doesn't quite look centered properly 
image

I also found the following on dark mode:

2. the edit icon is not very visible in dark mode

image

3. The `+` icon for the `Add a server` menu item is also not very visible in dark mode

image

**Review by matthewbirtch — APPROVED** [review]
Looks good @devinbinnie! We might need to figure out a solution for badges display on top of the server icon on the dropdown button as looks a little awkward when you have 2+ digits, but let's consider that for a potential iterative update

**Pull request merged**
